### PR TITLE
descriptive errmsg when hostpath dne

### PR DESCRIPTION
**Pull request by stefanrepl**
#### What type of PR is this?

type::feature

#### What this PR does / why we need it:
Shows a more descriptive error message when a user chooses a host path that does not exist.  It is needed so our users/vendors can easily understand what they are doing wrong.

#### Which issue(s) this PR fixes:
Fixes #

#### Special notes for your reviewer:
Attached screenshots show the two separate views a user might see this.  One is updating  storage and one creating storage.

#### Does this PR introduce a user-facing change?

```release-note
     Display descriptive error when provided host path does not exist.
```

#### Does this PR require documentation?

NONE
Screen Shot 2022-01-13 at 9 46 04 AM
Screen Shot 2022-01-13 at 9 46 14 AM

### CODE REVIEW

File: pkg/snapshot/filesystem.go [hunk 1]
```diff
@@ -52,6 +52,14 @@ type ResetFileSystemError struct {
 	Message string
 }
 
+type HostPathFileSystemError struct {
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem.go:55]: how about calling it
```suggestion
type HostPathNotFoundError struct {
```

File: pkg/handlers/snapshots.go [hunk 2]
```diff
@@ -431,6 +444,11 @@ func configureMinioFileSystemProvider(ctx context.Context, clientset kubernetes.
 		ForceReset:       fileSystemOptions.ForceReset,
 		FileSystemConfig: fileSystemOptions.FileSystemConfig,
 	}
+
+	if _, err := os.Stat(*deployOptions.FileSystemConfig.HostPath); os.IsNotExist(err) {
```
Comment (sgalsaleh) [inline pkg/handlers/snapshots.go:448]: would be a good idea to catch other errors as well, something like
```go
if _, err := os.Stat(*deployOptions.FileSystemConfig.HostPath); os.IsNotExist(err) {
    return &kotssnapshot.HostPathFileSystemError{Message: "provided host path does not exist"}
} else if err != nil {
    return errors.Wrap(err, "failed to os stat")
}
```

File: pkg/snapshot/filesystem.go [hunk 3]
```diff
@@ -52,6 +52,14 @@ type ResetFileSystemError struct {
...
+	Message string
+}
+
+func (e HostPathFileSystemError) Error() string {
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem.go:59]: nit pick: let's keep the type definition and the `Error()` function implementation coupled for the `ResetFileSystemError` type. something like:
```go
type ResetFileSystemError struct {
	Message string
}

func (e ResetFileSystemError) Error() string {
	return e.Message
}

type HostPathNotFoundError struct {
	Message string
}

func (e HostPathNotFoundError) Error() string {
	return e.Message
}
```

File: pkg/handlers/snapshots.go (hunk 2 above)
Comment (stefanrepl) [reply pkg/handlers/snapshots.go:448]: ah ofcourse, addressed

File: pkg/snapshot/filesystem.go (hunk 1 above)
Comment (stefanrepl) [reply pkg/snapshot/filesystem.go]: sounds good, changed

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**